samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java

```diff
@@ -1249,6 +1254,89 @@ public String startMoveQuestion()
 
+  public String checkSolution() {
+		setOutComeParams();
+		getCheckedQuestion();
+		GradingService delegate = new GradingService();
+		Map<Integer, String> answersMap = new HashMap<Integer, String>();
+		LinkedHashMap<String, String> answersMapValues = new LinkedHashMap<String, String>();
+		ItemDataIfc item = (ItemDataIfc) this.currentItems.get(0);
+
+		Random random = new Random();
+		long randomGradingId = Long.valueOf(Math.abs(random.nextInt()));
+		List<List<String>> texts = delegate.extractCalcQAnswersArray(answersMap, answersMapValues, item, randomGradingId, this.getAgentId());
+
+		//changing solutions ex: {{w}} with numbers
+		delegate.replaceSolutionOnFeedbackWithNumbers(answersMapValues, item, texts);
+
+		ItemContentsBean itemBean = new ItemContentsBean();
+		List<ItemGradingData> datas = new ArrayList();
+		String keysString = "";
+		int i = 0;
+
+		ItemTextIfc text = (ItemTextIfc) item.getItemTextArraySorted().toArray()[0];
+		List<FinBean> fins = new ArrayList<FinBean>();
+		List<AnswerIfc> calcQuestionEntities = text.getAnswerArraySorted();
+		Iterator<AnswerIfc> iter = calcQuestionEntities.iterator();
+		while (iter.hasNext()) {
+			AnswerIfc answer = iter.next();
+
+			// Checks if the 'answer' object is a variable or a real answer
+			if(delegate.extractVariables(answer.getText()).isEmpty()){
+				continue;
+			}
+
+			ItemGradingData data = new ItemGradingData();
+			String answerKey = (String)answersMapValues.get(answer.getLabel());
+			int decimalPlaces = Integer.valueOf(answerKey.substring(answerKey.indexOf(',')+1, answerKey.length()));
+			answerKey = answerKey.substring(0, answerKey.indexOf("|")); // cut off extra data e.g. "|2,3"
+			//We need the key formatted in scientificNotation
+			answerKey = delegate.toScientificNotation(answerKey, decimalPlaces);
+			keysString = keysString.concat(answerKey + ", ");
+
+			data.setAnswerText(answerKey);
+			data.setIsCorrect(true);
+			data.setPublishedAnswerId(answer.getId());
+			datas.add(data);
+
+			FinBean fbean = new FinBean();
+			fbean.setItemContentsBean(itemBean);
+			fbean.setAnswer(answer);
+			fbean.setText((String) texts.get(0).toArray()[i++]);
+			fbean.setHasInput(Boolean.TRUE); // input box
+			fbean.setItemGradingData(data);
+			fbean.setResponse(ComponentManager.get(FormattedText.class).convertFormattedTextToPlaintext(data.getAnswerText()));
+			fbean.setIsCorrect(true);
+			fins.add(fbean);
+		}
+
+		if (keysString.length() > 2) {
+			  keysString = keysString.substring(0, keysString.length()-2); // truncating the comma and blank on the end
+		}
+
+		itemBean.setItemData(item);
+		itemBean.setPointsForEdit(item.getScore().toString());
+		itemBean.setMaxPoints(item.getScore());
+		itemBean.setFeedbackValue(item.getCorrectItemFeedbackValue());
+		itemBean.setInfeedbackValue(item.getInCorrectItemFeedbackValue());
+		itemBean.setKey(keysString);
+		itemBean.setItemGradingDataArray(datas);
+
+		FinBean fbean = new FinBean();
+		if (texts.toArray().length>i)
```

Comment:
having spaces around the > will make this easier to read. Also, not using brackets on the if can cause problems later when people add more code to the blocks.